compiler/noirc_frontend/src/parser/parser/modifiers.rs

```diff
@@ -0,0 +1,38 @@
+use noirc_errors::Span;
+
+use crate::{ast::ItemVisibility, token::Keyword};
+
+use super::Parser;
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub(crate) struct Modifiers {
+    pub(crate) visibility: ItemVisibility,
+    pub(crate) visibility_span: Span,
+    pub(crate) unconstrained: Option<Span>,
+    pub(crate) comptime: Option<Span>,
+    pub(crate) mutable: Option<Span>,
+}
+
+impl<'a> Parser<'a> {
+    /// Modifiers = 'unconstrained'? ItemVisibility 'comptime'? 'mut'?
+    pub(crate) fn parse_modifiers(&mut self, allow_mutable: bool) -> Modifiers {
+        let unconstrained = self.parse_modifier(Keyword::Unconstrained);
+
+        let start_span = self.current_token_span;
+        let visibility = self.parse_item_visibility();
+        let visibility_span = self.span_since(start_span);
+
+        let comptime = self.parse_modifier(Keyword::Comptime);
+        let mutable = if allow_mutable { self.parse_modifier(Keyword::Mut) } else { None };
```

Comment:
When do we allow `mut` in the same position we allow `unconstrained`, `pub`, or `comptime`?

Reply:
`pub mut comptime global Foo = 0;` is the only location I think.